pkg/util/kubernetes/hpa/datadogexternal.go

```diff
@@ -31,42 +31,79 @@ func init() {
 	datadogStats.Set("Queries", datadogQueries)
 }
 
+type Metric struct {
```

Comment:
`Metric` is not a great name here. We should either add the metric name in here instead of keeping track of it outside, or rename the type to `Point` or something.